terraform/constellation-cluster/variables.tf

```
@@ -0,0 +1,89 @@
+variable "csp" {
+  type        = string
+  description = "The CSP to create the cluster in"
+  validation {
+    condition     = var.csp == "aws" || var.csp == "azure" || var.csp == "gcp"
+    error_message = "The CSP must be 'aws', 'azure' or 'gcp'."
```

```suggestion
    condition     = var.csp == "aws" 
    error_message = "The CSP must be 'aws'."
```

Add others once implemented to prevent confusion.